paymentsheet/src/test/java/com/stripe/android/utils/RecordingLinkPaymentLauncher.kt

```diff
@@ -0,0 +1,80 @@
+package com.stripe.android.utils
+
+import androidx.activity.result.ActivityResultCaller
+import app.cash.turbine.ReceiveTurbine
+import app.cash.turbine.Turbine
+import com.stripe.android.link.LinkActivityResult
+import com.stripe.android.link.LinkConfiguration
+import com.stripe.android.link.LinkPaymentLauncher
+import org.mockito.kotlin.any
+import org.mockito.kotlin.doAnswer
+import org.mockito.kotlin.mock
+
+object RecordingLinkPaymentLauncher {
+    fun noOp(): LinkPaymentLauncher {
+        return mock()
+    }
+
+    suspend fun test(test: suspend Scenario.() -> Unit) {
+        val registerCalls = Turbine<RegisterCall>()
+        val unregisterCalls = Turbine<Unit>()
+        val presentCalls = Turbine<PresentCall>()
+
+        val launcher = mock<LinkPaymentLauncher> {
+            on { register(any<ActivityResultCaller>(), any()) } doAnswer { invocation ->
+                val arguments = invocation.arguments
+
+                registerCalls.add(
+                    @Suppress("UNCHECKED_CAST")
+                    RegisterCall(
+                        activityResultCaller = arguments[0] as ActivityResultCaller,
+                        callback = arguments[1] as (LinkActivityResult) -> Unit,
+                    )
+                )
+            }
+
+            on { unregister() } doAnswer {
+                unregisterCalls.add(Unit)
+            }
+
+            on { present(any()) } doAnswer { invocation ->
+                val arguments = invocation.arguments
+
+                presentCalls.add(
+                    PresentCall(
+                        configuration = arguments[0] as LinkConfiguration,
+                    )
+                )
+            }
+        }
```

Comment:
`LinkPaymentLauncher` is a final class so which I decided not to refactor in this PR. This test utility creates the mock internally so that it can be replaced in the future by a test implementation if we refactor the class.